lib/common/model.js

```diff
@@ -260,6 +260,14 @@ function ModelFactory (
                 return self.runNativeMongo('findAndModify', _arguments);
             }
 
+            if (update.$pullAll) {
+                if (!update.$set) {
+                    update.$set = {};
+                }
+                this.addWaterlineCreatedAtToObject(update, '$set');
+                return self.runNativeMongo('findAndModify', _arguments);
+            }
+
```

Comment:
The above three code blocks look quite similar. Is it possible to combine them to a simpler one?